bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java

```diff
@@ -308,38 +332,177 @@ private HostAddress getHostAddress(long id, CaseDbConnection connection) throws
+		} 
+	}
+	
+	/**
+	 * Adds a row to the host address dns ip map table.
+	 *
+	 * @param dnsNameAddress The DNS name.
+	 * @param ipAddress      An IP address associated with the DNS name.
+	 * @param time           Timestamp when this relationship was true.
+	 * @param source         The source.
+	 * @param connection     The db connection. Null is not permitted.
+	 *
+	 * @throws TskCoreException
+	 */
+	private void addHostNameAndIpMapping(HostAddress dnsNameAddress, HostAddress ipAddress, Long time, Content source, final CaseDbConnection connection) throws  SQLException {
 
 		if (dnsNameAddress.getAddressType() != HostAddress.HostAddressType.HOSTNAME) {
 			throw new IllegalArgumentException("A host name address is expected.");
 		}
 		if ((ipAddress.getAddressType() != HostAddress.HostAddressType.IPV4) && (ipAddress.getAddressType() != HostAddress.HostAddressType.IPV6)) {
 			throw new IllegalArgumentException("An IPv4/IPv6 address is expected.");
 		}
+		if (Objects.isNull(connection)) {
+			throw new IllegalArgumentException("null connection passed to addHostNameAndIpMapping");
+		}
 
 		String insertSQL = db.getInsertOrIgnoreSQL(" INTO tsk_host_address_dns_ip_map(dns_address_id, ip_address_id, source_obj_id, time) "
 				+ " VALUES(?, ?, ?, ?) ");
 
-		db.acquireSingleUserCaseWriteLock();
-		try (CaseDbConnection connection = this.db.getConnection()) {
+		PreparedStatement preparedStatement = connection.getPreparedStatement(insertSQL, Statement.NO_GENERATED_KEYS);
 
-			PreparedStatement preparedStatement = connection.getPreparedStatement(insertSQL, Statement.NO_GENERATED_KEYS);
+		preparedStatement.clearParameters();
+		preparedStatement.setLong(1, dnsNameAddress.getId());
+		preparedStatement.setLong(2, ipAddress.getId());
+		preparedStatement.setLong(3, source.getId());
+		if (time != null) {
+			preparedStatement.setLong(4, time);
+		} else {
+			preparedStatement.setNull(4, java.sql.Types.BIGINT);
+		}
+		connection.executeUpdate(preparedStatement);
+		recentHostNameAndIpMappingCache.put(ipAddress.getId(), new Byte((byte)1));
+		recentHostNameAndIpMappingCache.put(dnsNameAddress.getId(), new Byte((byte)1));
+	}
 
-			preparedStatement.clearParameters();
-			preparedStatement.setLong(1, dnsNameAddress.getId());
-			preparedStatement.setLong(2, ipAddress.getId());
-			preparedStatement.setLong(3, source.getId());
-			if (time != null) {
-				preparedStatement.setLong(4, time);
-			} else {
-				preparedStatement.setNull(4, java.sql.Types.BIGINT);
+	/**
+	 * Returns true if addressObjectId is used as either IP or host name
+	 * <br>
+	 * <b>Note:</b> This api call uses a database connection. Do not invoke within a transaction.
+	 * 
+	 * @param address
+	 * @return 
+	 */
+	public boolean existsHostNameAndIpMapping(long addressObjectId) throws TskCoreException {
```

Comment:
I think it would be more natural if this method were named hostNameAndIpMappingExists().

Reply:
renamed